doc/api/url.md

```diff
@@ -2,139 +2,244 @@
+    (`/`), then the literal string '/' is appended to `result`.
+  * The value of `urlObject.pathname` is appended to `result`.
+* Otherwise, if `urlObject.pathname` is not `undefined` and is not a string, an
+  [`Error`][] is thrown.
+* If the `urlObject.search` property is `undefined` and if the `urlObject.query`
+  property is an `Object`, the literal string `?` is appended to `result`
+  followed by the output of calling the `querystring` module's `stringify()`
+  function passing the value of `urlObject.query`.
+* Otherwise, if `urlObject.search` is a string:
+  * If the value of `urlObject.search` *does not start* with the ASCII question
+    mark (`?`) character, the literal string `?` is appended to `result`.
+  * The value of `urlObject.search` is appended to `result`.
+* Otherwise, if `urlObject.search` is not `undefined` and is not a string, an
+  [`Error`][] is thrown.
+* If the `urlObject.hash` property is a string:
+  * If the value of `urlObject.hash` *does not start* with the ASCII hash (`#`)
+    character, the literal string `#` is appended to `result`.
+  * The value of `urlObject.hash` is appended to `result`.
+* Otherwise, if the `urlObject.hash` property is not `undefined` and is not a
+  string, an [`Error`][] is thrown.
+* `result` is returned.
+
 
 ## url.parse(urlStr[, parseQueryString][, slashesDenoteHost])
 <!-- YAML
 added: v0.1.25
 -->
 
-Take a URL string, and return an object.
-
-Pass `true` as the second argument to also parse the query string using the
-`querystring` module. If `true` then the `query` property will always be
-assigned an object, and the `search` property will always be a (possibly
-empty) string. If `false` then the `query` property will not be parsed or
-decoded. Defaults to `false`.
-
-Pass `true` as the third argument to treat `//foo/bar` as
-`{ host: 'foo', pathname: '/bar' }` rather than
-`{ pathname: '//foo/bar' }`. Defaults to `false`.
+* `urlStr` {string} The URL string to parse.
+* `parseQueryString` {boolean} If `true`, the `query` property will always
+  be set to an object returned by the `querystring` module's `parse()` method,
+  and the `search` property will always be a string. If `false`, the `query`
+  property on the returned URL object will be an unparsed, undecoded string.
+  Defaults to `false`.
+* `slashesDenoteHost` {boolean} If `true`, the first token after the literal
+  string `//` and preceeding the next `/` will be interpreted as the `host`.
+  For instance, given `//foo/bar`, the result would be
+  `{host: 'foo', pathname: '/bar'}` rather than `{pathname: '//foo/bar'}`.
+  Defaults to `false`.
+
+The `url.parse()` method takes a URL string, parses it, and returns a URL
+object.
 
 ## url.resolve(from, to)
 <!-- YAML
 added: v0.1.25
 -->
 
-Take a base URL, and a href URL, and resolve them as a browser would for
-an anchor tag.  Examples:
+* `from` {string} The HREF URL being resolved.
+* `to` {string} The Base URL being resolved against.
```

Comment:
Isn't this the other way around? You're starting with `from` (your base) and then resolving `to`.